File: backend/src/main/java/com/inventory/dto/ProductRawMaterialDTO.java
package com.inventory.dto;

import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Positive;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class ProductRawMaterialDTO {
    private Long id;
    
    @NotNull(message = "Raw material ID is required")
    private Long rawMaterialId;
    
    private String rawMaterialName;
    
    @NotNull(message = "Quantity required is required")
    @Positive(message = "Quantity required must be positive")
    private Integer quantityRequired;
}

File: backend/src/main/kotlin/com/inventory/dto/ProductRawMaterialDTO.kt
package com.inventory.dto

import jakarta.validation.constraints.NotNull
import jakarta.validation.constraints.Positive

data class ProductRawMaterialDTO(
    var id: Long? = null,
    
    @field:NotNull(message = "Raw material ID is required")
    var rawMaterialId: Long? = null,
    
    var rawMaterialName: String? = null,
    
    @field:NotNull(message = "Quantity required is required")
    @field:Positive(message = "Quantity required must be positive")
    var quantityRequired: Int = 0
)
